### ISSUE TITLE
wrapX support for ol.source.Image and subclasses, including ol.source.ImageVector

### ISSUE THREAD

**Issue by big8extreme** [openlayers/openlayers#4708]
ImageVector doesn't wrap over the map.

**Comment by tschaub** [comment]
Can you provide links to a minimal example that demonstrates the issue?  A [gist](https://gist.github.com/) with the minimal code would also do.  A screenshot would even help.

**Comment by big8extreme** [comment]
this exemple expose the problem:
you can see that the image generate with imageVector dont wrap over the map, even if we set wrapX to true.

http://openlayers.org/en/master/examples/image-vector-layer.html
![image vector layer - google chrome_2](https://cloud.githubusercontent.com/assets/1871677/12546872/113e6b2a-c34d-11e5-9326-a8e29e818e6a.jpg)

```javascript
new ol.layer.Image({
    source: new ol.source.ImageVector({
        source: new ol.source.Vector({
            url: 'data/geojson/countries.geojson',
            format: new ol.format.GeoJSON()
        }),
        style: new ol.style.Style({
            fill: new ol.style.Fill({
                color: 'rgba(255, 255, 255, 0.6)'
            }),
            stroke: new ol.style.Stroke({
                color: '#319FD3',
                width: 1
            })
        })
    })
})
```

**Comment by big8extreme** [comment]
up the problem

**Comment by ahocevar** [comment]
There is no `wrapX` option for `ol.source.Image` and its subclasses. This means that no x-wrapping is available for `ol.source.ImageVector`.

**Comment by ahocevar** [comment]
If someone is interested in implementing `wrapX` for `ol.source.Image` and subclasses, we'd accept pull requests.

**Comment by jgalrito-i9** [comment]
This would be a much needed requirement for a project I'm working on, since there are massive performance gains during interactions with layers with thousands of features on ImageVector sources vs Vector layers. I would pick this up but I'm not that familiar with the internals of OpenLayers to even know where to start, but maybe if someone can point me in the right direction I could try to do something.

**Comment by ahocevar** [comment]
Fixes for image vector with #7501.

**Comment by yudianjob** [comment]
使用ImageVector缩放地图时，不会显示所有feature，zoom越大显示越多，有没有什么办法控制显示所有feature呢
